mitiq/tests/test_folding_pyquil.py

```diff
@@ -1,48 +1,112 @@
 import numpy as np
 import random
 from pyquil import Program
-from pyquil.gates import X, Y, Z
-from mitiq.folding_pyquil import local_folding, unitary_folding, sampling_stretcher, left_stretcher
+from pyquil.gates import X, Y, Z, CNOT
+from pyquil.unitary_tools import program_unitary
+from mitiq.folding_pyquil import (
+    local_folding,
+    unitary_folding,
+    sampling_stretcher,
+    left_stretcher,
+)
+from copy import deepcopy
+
+
+STRETCH_VALS = [1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0]
+DEPTH = 50
+
+# NUM_SHOTS must be 1 in order to pass the next tests.
+# This is due to the fact that num_shots are not retained by
+# some methods of the Program class (bug?)
+NUM_SHOTS = 1
 
 
 def random_circuit(depth: int):
-    """Returns a single-qubit random circuit based on Pauli gates."""
+    """Returns a 2-qubit random circuit based on a simple gate set."""
     prog = Program()
+    gate_set = [X(0), Y(0), Z(0), X(1), Y(1), Z(1), CNOT(0, 1), CNOT(1, 0)]
     for _ in range(depth):
-        prog += random.choice([X(0), Y(0), Z(0)])
+        prog += random.choice(gate_set)
+    prog.num_shots = NUM_SHOTS
```

Comment:
Should this be `prog.wrap_in_numshots_loop`? Is there a difference?

Reply:
In practice it's the same. But I used your suggestion because `wrap_in_numshots_loop` looks more similar to what a user would do in a real scenario.